### PR TITLE
USHIFT-1394: Create volume group only when needed

### PR DESCRIPTION
**Pull request by pacevedom**

### CODE REVIEW

**Comment by pacevedom** [comment]
/pj-rehearse pull-ci-openshift-microshift-main-microshift-metal-tests pull-ci-openshift-microshift-main-microshift-e2e pull-ci-openshift-microshift-main-e2e-openshift-conformance-reduced

**Comment by dhellmann** [comment]
@pacevedom @eggfoobar can you add descriptions to the commits so someone looking at the history understands _why_ this change was made?

**Comment by pacevedom** [comment]
/pj-rehearse ack
Failure is unrelated to volumes. Volume groups were created in both metal-host VMs and ec2 VMs and microshift greenboot checks worked.
Ready for review.

File: ci-operator/step-registry/openshift/microshift/infra/volume-create/openshift-microshift-infra-volume-create-commands.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  ServerAliveCountMax 1200
+EOF
+chmod 0600 "${HOME}/.ssh/config"
+
+cat > "${HOME}"/volume-create.sh <<EOF
+#!/bin/bash
+set -xeuo pipefail
+
+sudo dnf install -y lvm2 jq
+pv_location=\$(sudo lsblk -Jd | jq -r '.blockdevices[] | select(.size == "200G") | "/dev/\(.name)"')
```
Comment (dhellmann) [inline ci-operator/step-registry/openshift/microshift/infra/volume-create/openshift-microshift-infra-volume-create-commands.sh:29]: In the course of working on the existing steps, I found the use of here-documents to generate script files introduced a lot of opportunities for errors when not escaping the right `$`. Can we put scripts like this into their own files here in this repo and then just copy them to the remote host to run them?

Comment (copejon) [reply ci-operator/step-registry/openshift/microshift/infra/volume-create/openshift-microshift-infra-volume-create-commands.sh:29]: The CI automation is looking for exactly one script file per step-ref, so there's not any good way to have a library of files per step. Everything has to be included in the `*-commands.sh` file alone.

In this sub-script's case, suppress variable expansion & substitution by quoting the delimiter (`"EOF"`).  There aren't any variables that need to be expanded in the current env anyway.

File: ci-operator/step-registry/openshift/microshift/infra/volume-create/openshift-microshift-infra-volume-create-commands.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+Host ${IP_ADDRESS}
+  User ${HOST_USER}
+  IdentityFile ${CLUSTER_PROFILE_DIR}/ssh-privatekey
+  StrictHostKeyChecking accept-new
+  ServerAliveInterval 30
+  ServerAliveCountMax 1200
+EOF
+chmod 0600 "${HOME}/.ssh/config"
+
+cat > "${HOME}"/volume-create.sh <<EOF
```
Comment (copejon) [inline ci-operator/step-registry/openshift/microshift/infra/volume-create/openshift-microshift-infra-volume-create-commands.sh:24]:
```suggestion
cat > "${HOME}"/volume-create.sh <<"EOF"
```

**Comment by dhellmann** [comment]
/pj-rehearse pull-ci-openshift-microshift-main-microshift-metal-tests

**Comment by dhellmann** [comment]
/pj-rehearse pull-ci-openshift-microshift-main-microshift-metal-tests

Last run failed with the dnf cache issue.

**Comment by copejon** [comment]
Is this PR still relevant or will the work shifting non-ostree tests to the test harness make it obsolete?

**Pull request closed without merging**